### PR TITLE
fix(TDI-47695): [JDBC TCK]: Allow different ui for different connector run env like studio component in job/studio metadata/cloud

### PR DESCRIPTION
**Pull request by wwang-talend**
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant 

### Why this PR is needed?
https://jira.talendforge.org/browse/TDI-47695

### What does this PR adds (design/code thoughts)?
https://jira.talendforge.org/browse/TDI-47695

### CODE REVIEW

File: component-form/component-form-core/src/main/java/org/talend/sdk/component/form/internal/converter/impl/widget/AbstractWidgetConverter.java [hunk 1]
```diff
@@ -307,6 +308,19 @@ protected Map<String, Collection<Object>> createCondition(final PropertyContext<
                             !Boolean.parseBoolean(ctx.getProperty().getMetadata().getOrDefault(negateKey, "false"));
                     final String strategy =
                             ctx.getProperty().getMetadata().getOrDefault(strategyKey, "default").toLowerCase(ROOT);
+
+                    if ("ui.scope".equals(paramRef)) {
```
[the remark this answers is not in the document]
Comment (wwang-talend) [reply component-form/component-form-core/src/main/java/org/talend/sdk/component/form/internal/converter/impl/widget/AbstractWidgetConverter.java:312]: can't ref component-api in this module project

File: component-api/src/main/java/org/talend/sdk/component/api/configuration/condition/UIScope.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    String STUDIO_SCOPE = "studio";
+
+    String CLOUD_SCOPE = "cloud";
+
+    String STUDIO_COMPONENT_SCOPE = "studio.component";
+
+    String STUDIO_CONNECTION_COMPONENT_SCOPE = "studio.component.connection";
+
+    String STUDIO_METADATA_SCOPE = "studio.metadata";
+
```
[the remark this answers is not in the document]
Comment (wwang-talend) [reply component-api/src/main/java/org/talend/sdk/component/api/configuration/condition/UIScope.java:31]: "STUDIO" for that

File: component-api/src/main/java/org/talend/sdk/component/api/configuration/condition/UIScope.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.api.configuration.condition;
+
+public interface UIScope {
```
[the remark this answers is not in the document]
Comment (wwang-talend) [reply component-api/src/main/java/org/talend/sdk/component/api/configuration/condition/UIScope.java:18]: not sure PlatformContext is a good idea, in fact, PlatformContext and ActiveIf do the similar work for hide some option for some case. one by (expression and compare function), one by the platform model show. 

And also as they both for hiding, not easy to process the logic for "show if", for example, consider this case:
```
@PlatformContext("cloud")
@Option
private boolean a;

@Option
@PlatformContext("cloud")
@ActiveIf(target="a", value="true")
private boolean b;

@Option
@PlatformContext("cloud")
@ActiveIf(target="b", value="true")
private boolean c;
```
for me, is easy to make thing wrong for show_if logic, so in my view, better to use "activeif" way.

File: component-api/src/main/java/org/talend/sdk/component/api/configuration/condition/UIScope.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (undx) [reply component-api/src/main/java/org/talend/sdk/component/api/configuration/condition/UIScope.java:31]: Ok, clear! Maybe just add a comment specifying it...

File: component-form/component-form-core/src/main/java/org/talend/sdk/component/form/internal/converter/impl/widget/AbstractWidgetConverter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (undx) [reply component-form/component-form-core/src/main/java/org/talend/sdk/component/form/internal/converter/impl/widget/AbstractWidgetConverter.java:312]: hum, you're right ;-)

**Review by undx — APPROVED** [review]
lgtm

**Pull request merged**